2a_grass/mixed_pfts.Rmd

```diff
@@ -0,0 +1,135 @@
+# Modify initial pools for mixed-PFT simulations
+
+Context: To test approaches to simulating mixed PFT management using the
+(single-PFT-at-a-time) Sipnet model, we want to generate simulations of the
+same site using two different PFTS, then experiment with post-processing them
+in different ratios.
+
+One challenge is that initial site conditions are set using remote sensed
+observations that are converted to estimates of pool size using PFT-specific
+assumptions. To convert an IC file to one for a different PFT, we also need
+to begin a switch from setting ICs based on remote sensing to setting them
+based on counterfactual assumptions of what remote sensing _would_ show if
+this site were actually mixed PFTs.
+
+Here I mostly do not address the need for counterfactuals and try to convert
+very simply by starting from sites known to be woody and then converting them
+to grass by:
+
+* Keeping soil C and initial soil water identical
+* Treating sensed LAI as if it were 100% from the PFT to be simulated
+	(this will probably overestimate LAI for all PFTs)
+* Setting sensed woody biomass to zero when simulating a nonwoody PFT
+
+which fields of the IC file to alter?
+	- abvGrdWood and wood_carbon_content = 0 for nonwoody pfts
+	- LAI two options:
+		- leave as-is, accept all PFTs start with too many leaves
+		- scale to area coverage (need to know area coverage at IC setup time)
+		==> I'm leaving as-is
+	- leaf carbon: rescale from LAI using PFT-specific SLA and leafC
+
+To generate site info:
+
+```{r}
+library(tidyverse)
+site_info <- read.csv("site_info.csv", colClasses = c(field_id = "character"))
+tree_sites <- site_info |>
+	filter(site.pft == "temperate.deciduous")
+tree_sites_as_grass <- tree_sites |>
+	mutate(site.pft == "grass",
+	id = paste0(id, "_grass")
+
+write.csv(tree_sites, "site_info_tree.csv", row.names = FALSE)
+write.csv(tree_sites_as_grass, "site_info_tree_as_grass.csv", row.names = FALSE)
+write.csv(
+	bind_rows(tree_sites, tree_sites_as_grass),
+	"site_info_tree_as_both.csv",
+	row.names = FALSE
+)
+```
+
+To generate new IC files taking new draws from all estimated distributions
+
+Note we only regenerate for grass -- tree runs keep using orig file.
+Note also that this adds new entries to the IC data dir (`data/IC_prep/`) for
+all `<id>_grass` sites. You can verify in `data/IC_prep/IC_means.csv` that the
+estimated distributions are the same as for the tree sites -- we change the draws,
+not the distributions of estimated true state.
+
+```{sh}
+./02_ic_build.R --site_info_path=site_info_tree_as_grass.csv \
+	--ic_outdir=IC_nowood_newdraws \
+	--additional_params='varname=wood_carbon_fraction,distn=norm,parama=0.0001,paramb=0.00001'
+mkdir IC_mix_newdraws
+cp IC_files/* IC_mix_newdraws
+cp IC_nowood_newdraws/* IC_mix_newdraws
+```
+
+The big caveat here is that is generates new draws of soilC and initial soil water for each ensemble member.
```

Comment:
Would the issue of having changing draws be avoided by grass LAI to 0? Not sure if I am missing something, but is there a reason for a trivally small leaf carbon content value?

Reply:
👍 I'll remove this whole section